reactive/kubernetes_master.py

```diff
@@ -2818,3 +2819,32 @@ def api_server_stopped():
     aws_iam = endpoint_from_flag('endpoint.aws-iam.available')
     if aws_iam:
         aws_iam.set_api_server_status(False)
+
+
+@when('cluster-dns.joined')
+def send_clusterdns_domain():
+    """Send domain to the Cluster DNS interface."""
+    cluster_dns = endpoint_from_flag('cluster-dns.joined')
+    dns_domain = hookenv.config('dns_domain')
+
+    hookenv.log('Sending dns_domain: {}'.format(dns_domain))
+    cluster_dns.send_domain(dns_domain)
+
+    set_flag('kubernetes-master.need-cluster-dns')
+    clear_flag('cluster-dns.joined')
```

Comment:
What if dns_domain changes? i.e. user runs `juju config kubernetes-master dns_domain=new.domain`. This handler is set up to only run once, so the new value will never be sent, right?

Reply:
I will have to take a look at that